### PR TITLE
GEN-1277: feat: added ProductReviewCommentsBlock

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Renames `services/averageRating.ts` --> `services/productReviews.ts`: It includes utility functions to fetch all things related with reviews - _average rating_ and _review comments_.
* Update `[[...slug]]` so we fetch latest review comments for product pages.
* Added a `ProductReviewCommentsBlock` - it will display latest reviews of a product.
* Promote `Rating` into a shareable component: It's the UI for the rating stars which is used in both `AverageRatingBlock` and `ProductReviewCommentsBlock` blocks.
* Update `InputSelect` so it support disabled options: I use this for _review comments filter by score_ feature for scores for which a product doesn't have any reviews.

https://github.com/HedvigInsurance/racoon/assets/19200662/464db38f-af20-4969-b458-ad620bf9e485

## Justify why they are needed

Part of _makers days_ initiative.

Can be tested for _Accident_ and _Car_ with the respective links:

* [Accident](https://hedvig-dot-com-git-makers-days-featprocut-review-6914ca-hedvig.vercel.app/se/forsakringar/olycksfallsforsakring/av)
* [Car](https://hedvig-dot-com-git-makers-days-featprocut-review-6914ca-hedvig.vercel.app/se/forsakringar/bilforsakring/av)

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3350** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3350" target="_blank">Graphite</a> 
    * **PR #3359** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3359" target="_blank">Graphite</a> 
      * **PR #3360** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3360" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3360?utm_source=stack-comment).

File: apps/store/src/components/Rating/Rating.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import styled from '@emotion/styled'
+import { theme } from 'ui'
+
+type Props = {
+  score: number
+}
+
+export const Rating = styled.div<Props>(({ score }) => ({
```
Comment (robinandeer) [inline apps/store/src/components/Rating/Rating.tsx:8]: Should we consider naming this more specifically? "Rating" for what?

`ProductReviewRating` might make more sense, no?

File: apps/store/src/blocks/ProductReviewCommentsBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        <Space y={2}>
+          <SpaceFlex direction="horizontal" align="center" space={1} wrap="wrap">
+            <InputSelect
+              name="product-review-filter-by-score"
+              options={filterOptions}
+              defaultValue={selectedScore}
+              onChange={handleFilterChange}
+            />
+
+            <Text color="textSecondary">Showing our latest reviews</Text>
```
Comment (robinandeer) [inline apps/store/src/blocks/ProductReviewCommentsBlock.tsx:75]: nit: lokalise

**Review by robinandeer — COMMENTED** [review]
Great stuff! Perhaps you can give a walkthrough

[](https://app.graphite.dev/macros/HedvigInsurance)

File: apps/store/src/components/Rating/Rating.tsx (hunk 1 above)
Comment (guilhermespopolin) [reply apps/store/src/components/Rating/Rating.tsx:8]: Yes, it does :) Changing it.

File: apps/store/src/blocks/ProductReviewCommentsBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  const filterOptions = useMemo<InputOptions>(() => {
+    if (!reviewComments) return []
+
+    const availableScores: Array<AvailableScore> = ['5', '4', '3', '2', '1']
+    const options: InputOptions = []
+    availableScores.forEach((score) => {
+      const commentsByScore = reviewComments.commentsByScore[score]
+      // It should not happen. Just being safe
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      if (commentsByScore == null) {
```
Comment (guilhermespopolin) [inline apps/store/src/blocks/ProductReviewCommentsBlock.tsx:32]: Use `===`

File: apps/store/src/blocks/ProductReviewCommentsBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  score: number
+  date: string
+  author: string
+  content: string
+}
+
+const Comment = ({ content, date, author, score }: CommentProps) => {
+  return (
+    <CommentWrapper>
+      <ProductRating style={{ '--size': '1.5rem' } as React.CSSProperties} score={score} />
```
Comment (guilhermespopolin) [inline apps/store/src/blocks/ProductReviewCommentsBlock.tsx:105]: Remove`--size: '1.5rem'`

**Review by robinandeer — APPROVED** [review]
Looks great to me, interested in seeing what you come up with in terms of custom data structures

Added via Giphy

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Oct 17, 10:36 AM**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3360).

**Pull request merged**